llvm/lib/Target/NVPTX/NVPTXLowerUnreachable.cpp

```diff
@@ -110,17 +110,25 @@ StringRef NVPTXLowerUnreachable::getPassName() const {
 }
 
 // =============================================================================
-// Returns whether a `trap` intrinsic should be emitted before I.
+// Returns whether a `trap` intrinsic would be emitted before I.
 //
 // This is a copy of the logic in SelectionDAGBuilder::visitUnreachable().
 // =============================================================================
 bool NVPTXLowerUnreachable::isLoweredToTrap(const UnreachableInst &I) const {
-  if (!TrapUnreachable)
-    return false;
-  if (!NoTrapAfterNoreturn)
-    return true;
   const CallInst *Call = dyn_cast_or_null<CallInst>(I.getPrevNode());
-  return Call && Call->doesNotReturn();
+  if (Call) {
```

Comment:
Nit: move the definition of `Call` into the `if` condition or [prefer early returns](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)